### PR TITLE
rework plugins discovery

### LINKED ISSUE #179: plugins probed even if not using entrypoints
I'm using aiohttp-s3-client and I'm getting pkg_resource not found.

I could give in and install setuptools, or set the AIOMISC_NO_PLUGINS variables, but I consider the automatic probing for plugins to be unnecessary.  It seems that PLUGINS are specifically tied to aiomisc.entrypoint() API, which aiohttp-s3-client doesn't use.

maybe aiomisc.entrypoint.entrypoint can run setup_plugins() before returning Entrypoint, that setup_plugins doesn't need to be called on import.

### PR DESCRIPTION
**Pull request by mosquito**
fixes #179

### CODE REVIEW

File: aiomisc/plugins/__init__.py [hunk 1]
```diff
@@ -1,24 +1,30 @@
...
+    logger = logging.getLogger(__name__)
 
-    for entry_point in pkg_resources.iter_entry_points("aiomisc.plugins"):
-        plugins[entry_point.name] = entry_point.load()
-
-    for entry_point in pkg_resources.iter_entry_points("aiomisc"):
-        plugins[entry_point.name] = entry_point.load()
+    for entry_point in chain(
+        entry_pont_iterator("aiomisc.plugins"),
+        entry_pont_iterator("aiomisc"),
```
Comment (leenr) [inline aiomisc/plugins/__init__.py:19]: I don't like this hardcode.
Maybe it's better to use `aiomisc.__name__` and `aiomisc.plugins.__name__` instead?

**Review by leenr — CHANGES_REQUESTED** [review]

File: pyproject.toml [hunk 2]
```diff
@@ -55,16 +55,17 @@ packages = [
...
 aiohttp-asgi = { version = "^0.5.2", optional = true }
 colorlog = "^6.0"
 croniter = { version = "^1.3.8", optional = true }
 logging-journald = [{ version = '*', platform = 'linux' }]
+python = "^3.7"
 raven = { version = "*", optional = true }
 rich = { version = "*", optional = true }
-typing_extensions = [{ version = '*', python = "< 3.10" }]
+setuptools = [{ version = '*', python = "< 3.8" }]
+typing_extensions = [{ version = '*', python = "< 3.8" }]
```
[the remark this answers is not in the document]
Comment (mosquito) [reply pyproject.toml:68]: The `typing.Protocol` requires this change.

File: aiomisc/plugins/__init__.py (hunk 1 above)
Comment (mosquito) [reply aiomisc/plugins/__init__.py:19]: `explicit is better than implicit`

File: aiomisc/compat.py [hunk 3]
```diff
@@ -29,6 +29,36 @@ def time_ns() -> int:
...
+# noinspection PyUnresolvedReferences
+try:
+    from importlib.metadata import Distribution, EntryPoint
+
+    def entry_pont_iterator(entry_point: str) -> Iterator[EntrypointProtocol]:
+        ep: EntryPoint
+        for dist in Distribution.discover():
+            for ep in dist.entry_points:
+                if ep.group == entry_point:
+                    yield ep        # type: ignore
```
[the remark this answers is not in the document]
Comment (mosquito) [reply aiomisc/compat.py:54]: `pkg_resources` is might be not installed on this stage

File: pyproject.toml [hunk 4]
```diff
@@ -55,16 +55,17 @@ packages = [
...
 aiohttp = { version = ">3", optional = true }
 aiohttp-asgi = { version = "^0.5.2", optional = true }
 colorlog = "^6.0"
 croniter = { version = "^1.3.8", optional = true }
 logging-journald = [{ version = '*', platform = 'linux' }]
+python = "^3.7"
 raven = { version = "*", optional = true }
 rich = { version = "*", optional = true }
-typing_extensions = [{ version = '*', python = "< 3.10" }]
+setuptools = [{ version = '*', python = "< 3.8" }]
```
[the remark this answers is not in the document]
Comment (mosquito) [reply pyproject.toml:67]: `pkg_resources` is part of `setuptools`

**Review by leenr — APPROVED** [review]

**Review by Alviner — APPROVED** [review]

**Pull request merged**